Formula/xapian.rb

```diff
@@ -19,8 +19,8 @@ class Xapian < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "e77c1412ab0f38e3acd480edbbf5a41a082ce3aad4db12ebc5cd287beccd0bc8"
   end
 
+  depends_on "python@3.10" => [:build, :test]
```

Comment:
I think this has a bunch of dependent formulae that are still on `python@3.9` and do make use of Xapian's Python bindings. Those will break with this change.

Reply:
I see, in that case we can keep `python@3.9` if we don't use `sphinx-doc` to provide `sphinx` and instead create a virtualenv at build time containing the module (i.e. we can vendor the `sphinx` module).

Reply:
Apologies for flip-flopping on this, but now that I've seen the changes required to stay on `python@3.9`: I'm kinda inclined to make the switch to `python@3.10`. We can switch it back if someone reports a dependent being broken...